### ISSUE TITLE
Splitting a pre trained model

### ISSUE THREAD

**Issue by abhi-rf** [facebookarchive/caffe2#826]
Hello everyone. I wanted to know how can I split a pre trained model , like squeezenet into two parts. Please suggest a way to do so !!! Thanks :) :)

**Comment by alex-jw-brooks** [comment]
Are you trying to split a pre-trained Caffe model, or a Caffe2 model? Also, what information are you trying to extract?

**Comment by abhi-rf** [comment]
I am trying to split a Caffe2 Model. Basicaly its one of my school mini projects. The first aim is I want to test if I split the model to let us say two sub models A and B , and then if I infer an image on these , let us say I pass it through A and then its output through B , will it show correct answer ?

**Comment by alex-jw-brooks** [comment]
Unfortunately, I have not tried to do anything like this with a Caffe2 model. :( I'm pretty new to both Caffe and Caffe2, but since nobody else is responding maybe I can at least help you come up with an idea, even if I don't have a solution myself. I recently did something similar converting a caffemodel into caffe2. 

In my pre-trained model, I had lots of issues converting to Caffe2 due to custom layers not addressed in the translator, as well as some dimensionality problems. It was ugly, but I got it to work by adding a couple of intermediate python layers as placeholders for the translator, and implementing those in Caffe2 to do the operation I needed. For example, in my network, I had something like 

someLayer -> SoftMaxWithLoss, which worked in Caffe, but broke in Caffe2 due to dimensional problems (since my network is fully convolutional). I got around it as follows
someLayer -> custom python operator -> Accuracy, where the custom operator simply resized as needed for my output layers. 

Do you know of any ways to relink your model in Caffe2 such that instead of 
input->output
you have
input->end_of_part_1 -> custom operator -> start_of_part_2 -> output? Then you could use the operator as your split point.

**Comment by abhi-rf** [comment]
Thanks for the effort :) :) But I only have caffe2. I dont want to convert from caffe to caffe2. I want to split the model structure into two parts

**Comment by abhi-rf** [comment]
I actually was not able to do it so I switched to PyTorch and viola!! It was easy there :) :) . Having my vacations soon so yeaa will figure it out on PyTorch to see if something can  be done in Caffe2. I did figure out a rough guideline on how it can be done and would keep you posted if it is successful. I guess you can try it on PyTorch as I was able to do it easily there.

Cheers
